### PR TITLE
Add data to transfer/onPayment

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/2092

### CODE REVIEW

File: src/neo/SmartContract/Native/Tokens/Nep17Token.cs [hunk 1]
```diff
@@ -67,10 +67,10 @@ internal protected virtual void Mint(ApplicationEngine engine, UInt160 account,
             state.Balance += amount;
             storage = engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_TotalSupply), () => new StorageItem(BigInteger.Zero));
             storage.Add(amount);
-            PostTransfer(engine, null, account, amount);
+            PostTransfer(engine, null, account, amount, data);
```
Comment (erikzhang) [inline src/neo/SmartContract/Native/Tokens/Nep17Token.cs:70]:
```suggestion
            PostTransfer(engine, null, account, amount, null);
```

We don't need data in Mint.

Comment (vncoelho) [reply src/neo/SmartContract/Native/Tokens/Nep17Token.cs:70]: For Nep17 not, maybe for NFT Mint yes.

File: src/neo/SmartContract/Native/Tokens/Nep17Token.cs [hunk 2]
```diff
@@ -85,7 +85,7 @@ internal protected virtual void Burn(ApplicationEngine engine, UInt160 account,
                 state.Balance -= amount;
             storage = engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_TotalSupply));
             storage.Add(-amount);
-            PostTransfer(engine, account, null, amount);
+            PostTransfer(engine, account, null, amount, data);
```
Comment (vncoelho) [inline src/neo/SmartContract/Native/Tokens/Nep17Token.cs:88]: Sounds good to have data here.

File: src/neo/SmartContract/Native/Tokens/Nep17Token.cs (hunk 1 above)
Comment (shargon) [reply src/neo/SmartContract/Native/Tokens/Nep17Token.cs]: Done, also was removed from Burn

**Review by erikzhang — APPROVED** [review]

File: src/neo/SmartContract/Native/Tokens/Nep17Token.cs (hunk 1 above)
Comment (vncoelho) [reply src/neo/SmartContract/Native/Tokens/Nep17Token.cs]: Maybe it should had been kept for `Burn`, no?

**Review by vncoelho — APPROVED** [review]

**Pull request merged**